tests/unit/api/main.py

```diff
@@ -427,7 +427,7 @@ def test_node_register_JSON_metadata(self):
                   "host": "ipmihost",
                   "user": "root",
                   "password": "tapeworm"}, metadata={
-                      "EK": {"val1": 1, "val2": 2}})
+                      "EK": json.dumps({"val1": 1, "val2": 2})})
```

Comment:
This needs to be done inside the API call, since the HTTP server is going to pass it the parsed JSON.